packages/fulfillment/src/services/fulfillment-module-service.ts

```diff
@@ -211,6 +227,74 @@ export default class FulfillmentModuleService<
     })
   }
 
+  @InjectManager("baseRepository_")
+  async retrieveFulfillment(
+    id: string,
+    config: FindConfig<FulfillmentTypes.FulfillmentDTO> = {},
+    @MedusaContext() sharedContext: Context = {}
+  ): Promise<FulfillmentTypes.FulfillmentDTO> {
+    const fulfillment = await this.fulfillmentService_.retrieve(
+      id,
+      config,
+      sharedContext
+    )
+
+    return await this.baseRepository_.serialize<FulfillmentTypes.FulfillmentDTO>(
+      fulfillment,
+      {
+        populate: true,
+      }
+    )
+  }
+
+  async retrieveFulfillmentOptions(
+    providerId: string
+  ): Promise<Record<string, any>[]> {
+    return await this.serviceProviderService_.getFulfillmentOptions(providerId)
+  }
+
+  @InjectManager("baseRepository_")
+  async listFulfillments(
+    filters: FulfillmentTypes.FilterableFulfillmentProps = {},
+    config: FindConfig<FulfillmentTypes.FulfillmentDTO> = {},
+    @MedusaContext() sharedContext: Context = {}
+  ): Promise<FulfillmentTypes.FulfillmentDTO[]> {
+    const fulfillments = await this.fulfillmentService_.list(
```

Comment:
What's the reason we have to override the list method here? Same with fulfillment options above?

Reply:
I do not override the method here, I did explicitly not generated the method for this entity as mention in the comment I ve put at the top of the file :)

Reply:
Got it, I missed that comment while reviewing 😄